### ISSUE TITLE
Incorrect total energy and energy components for batched VMC with drift and spinors

### ISSUE THREAD

**Issue by camelto2** [QMCPACK/qmcpack#4910]
**Describe the bug**

**TLDR** Batched + Drift shows a bias with spinor and normal code path.  Normal code path exhibits bug when doing a completely spin polarized calculation, i.e. only up electrons. 

In the process of doing some testing of the spinor code, we were trying to set up the exact same problem between a spinor wave function and a normal calculation without spin-orbit before moving to production VMC/DMC with spin-orbit. We ran QE with normal settings as well as noncolin = .true. both using AREP potentials. This should produce equivalent DFT calculations, just one using normal orbitals and one using spinors. These agree in energy at the DFT level and they should agree as well at the QMC level. 

The first test was checking that the VMC energy agreed between spinor and non spinor without jastrows. Those total energies agreed with the batched code, so everything looked reasonable. We then moved to a wave function that had jastrows. Running the spinor and non spinor case in the batched code both with the same jastrows had errors in the total energy. That was alarming, so we did the same run with the legacy code and it turned out that the energies agreed. So it seems there is a bug in the batched drivers somewhere. Then we reran the same tests, using batched/legacy and with/without jastrows but this time running without drift in the VMC drivers. If drift is not included, the spinor calculations seem to agree with the normal non spinor calculations. 

An example is shown below just showing energies and kinetic

w/ Jastrows:
```
> 
> vmc_legacy/spinor/drift  series 0
>   LocalEnergy           =          -6.98049 +/-          0.00026
>   Kinetic               =            2.6990 +/-           0.0015
> 
> vmc_legacy/spinor/nodrift  series 0
>   LocalEnergy           =         -6.980795 +/-         0.000088
>   Kinetic               =           2.69787 +/-          0.00067
>   
>   vmc_legacy/nospinor/drift  series 0
>   LocalEnergy           =          -6.98052 +/-          0.00028
>   Kinetic               =            2.7020 +/-           0.0017
> 
> vmc_legacy/nospinor/nodrift  series 0
>   LocalEnergy           =          -6.98065 +/-          0.00027
>   Kinetic               =            2.7036 +/-           0.002
>   
>   vmc_batched/spinor/drift  series 0
>   LocalEnergy           =          -6.98252 +/-          0.00049
>   Kinetic               =            2.6403 +/-           0.0024
> 
> vmc_batched/spinor/nodrift  series 0
>   LocalEnergy           =          -6.98059 +/-          0.00087
>   Kinetic               =            2.7011 +/-           0.0028
> 
> vmc_batched/nospinor/drift  series 0
>   LocalEnergy           =          -6.98050 +/-          0.00023
>   Kinetic               =            2.7016 +/-           0.0017
> 
> vmc_batched/nospinor/nodrift  series 0
>   LocalEnergy           =         -6.980612 +/-         0.000090
>   Kinetic               =           2.69838 +/-          0.00078
>
```

So currently, **with spinors and batched and jastrows and drift** has noticeable error, particularly with the kinetic energy. If I remove the jastrows, everything actually agrees for the spinor case.  
 

 
**To Reproduce**
1. last commit: 5bf03ee0409fca60acd214db8140222c66409745
2. normal make build, just using -DQMC_COMPLEX=1
3. running qmcpack as normal, i.e. qmcpack_complex qmc.in.xml 

**Expected behavior**
Energy should agree between batched drift, batched no drift, legacy drift, legacy no drift, regardless of whether there is a jastrow or not. 

**System:**
 - my M1 mac

**Additional context**
Working on adding files for reproducibility

**Comment by camelto2** [comment]
[completely_polarized_reproducer.tgz](https://github.com/QMCPACK/qmcpack/files/14091087/completely_polarized_reproducer.tgz)

**Comment by camelto2** [comment]
Ok, partial false alarm. I was running with the incorrect version of the code where I had been testing some changes to study the spinors. It turns out running with a clean develop **IS OK** for the non spinor case. Unfortunately, there is something weird with batched drift VMC using spinors if a jastrow is included. Without a jastrow works for spinors

**Comment by prckent** [comment]
I tweaked the title based on your comment. Please revise further if needed.

**Comment by camelto2** [comment]
also commented out the stuff that was wrong
